functions/v2/helloworld/hello_pubsub.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_helloworld_pubsub_v2]
+
+// Package helloworld provides a set of Cloud Functions samples.
+package helloworld
+
+import (
+	"log"
+)
+
+// A CloudEvent containing the Pub/Sub message.
+// See the documentation for more details:
+// https://cloud.google.com/eventarc/docs/cloudevents#pubsub
+type CloudEventMessage struct {
+	Message PubSubMessage
+}
+
+// PubSubMessage is the payload of a Pub/Sub event.
+// See the documentation for more details:
+// https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage
+type PubSubMessage struct {
+	Data []byte `json:"data"`
+}
+
+// HelloPubSub consumes a CloudEvent message and extracts the Pub/Sub message.
+func HelloPubSub(cem CloudEventMessage) error {
```

Comment:
issue: The Functions Framework uses `github.com/cloudevents/sdk-go/v2`, so we should go ahead and use cloudevents.Event. In fact, looking closer at the code, that's required in order to make a CloudEvent Function instead of a Background Function that's consuming a CloudEvent.

https://github.com/GoogleCloudPlatform/functions-framework-go#cloudevent-functions

* Delete CloudEventMessage type
* Add cloudevents SDK
* Use the cloudevent DataAs method to unmarshal to your defined PubSubMessage type

Reply:
Hrm, I followed the way that Ace did it in his sample: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/functions/v2/index.js

Reply:
Updated it to use the cloudevents sdk.  This makes it about 2x as big.